[ci] [R-package] builds with Intel compiler failing

## Description

Starting a few days ago, R-package builds with `icc` started failing like this:

```text
** libs
using C compiler: ‘Intel(R) oneAPI DPC++/C++ Compiler 2024.2.1 (2024.2.1.20240711)’
make[1]: Entering directory '/tmp/RtmpydpZRQ/R.INSTALL1ebe43e7cebc/rlang/src'
/opt/intel/oneapi/compiler/2024.2/bin/icx -std=gnu23 -I"/opt/R/devel-intel/lib/R/include" -DNDEBUG -I./rlang/  -I/usr/local/include   -fvisibility=hidden -fpic  -O3 -fp-model precise -Wall -Wstrict-prototypes  -c capture.c -o capture.o
touch internal.c
/opt/intel/oneapi/compiler/2024.2/bin/icx -std=gnu23 -I"/opt/R/devel-intel/lib/R/include" -DNDEBUG -I./rlang/  -I/usr/local/include   -fvisibility=hidden -fpic  -O3 -fp-model precise -Wall -Wstrict-prototypes  -c internal.c -o internal.o
/opt/intel/oneapi/compiler/2024.2/bin/icx -std=gnu23 -I"/opt/R/devel-intel/lib/R/include" -DNDEBUG -I./rlang/  -I/usr/local/include   -fvisibility=hidden -fpic  -O3 -fp-model precise -Wall -Wstrict-prototypes  -c rlang.c -o rlang.o
In file included from rlang.c:5:
In file included from ./rlang/rlang.c:12:
./rlang/dots-info.c:58:12: error: use of undeclared identifier 'R_DotsExist'
   58 |     return R_DotsExist(env);
      |            ^
./rlang/dots-info.c:78:22: error: use of undeclared identifier 'R_DotsLength'
   78 |     return (r_ssize) R_DotsLength(env);
      |                      ^
./rlang/dots-info.c:98:12: error: use of undeclared identifier 'R_DotsNames'
   98 |     return R_DotsNames(env);
...
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[1]: *** [/opt/R/devel-intel/lib/R/etc/Makeconf:190: rlang.o] Error 1
make[1]: Leaving directory '/tmp/RtmpydpZRQ/R.INSTALL1ebe43e7cebc/rlang/src'
ERROR: compilation failed for package ‘rlang’
```

## Reproducible example

This is happening on all PRs. Example: https://github.com/lightgbm-org/LightGBM/actions/runs/23982110692/job/70280655229?pr=7124

## Environment info

N/A

## Additional Comments

Noticing that the `R-devel` in those images is about a month and a half old:

```text
* using R Under development (unstable) (2026-02-19 r89439)
```

I suspect the image we use to get R + Intel compilers has not been published in a while.

--- a/.github/workflows/r_package.yml
+++ b/.github/workflows/r_package.yml
@@ -273,14 +273,16 @@ jobs:
         # references:
         #   * CRAN "additional checks": https://cran.r-project.org/web/checks/check_issue_kinds.html
         #   * images: https://r-hub.github.io/containers/containers.html
+        #
+        # Generally, only images that are still supported, listed at
+        # https://r-hub.github.io/containers/#available-containers, should be used.
         image:
-          # unless a specific reason to do otherwise, just test
-          # oldest and newest supported versions of compilers here
-          - clang16
+          - clang21
           - clang22
-          - gcc14
           - gcc16
           - rchk
+          - r-devel-linux-x86_64-debian-clang
+          - r-devel-linux-x86_64-debian-gcc
     runs-on: ubuntu-latest
     container: ghcr.io/r-hub/containers/${{ matrix.image }}:latest
     steps:

--- a/.github/workflows/r_package.yml
+++ b/.github/workflows/r_package.yml
@@ -281,8 +281,8 @@ jobs:
           - clang22
           - gcc16
           - rchk
-          - r-devel-linux-x86_64-debian-clang
-          - r-devel-linux-x86_64-debian-gcc
+          - ubuntu-clang
+          - ubuntu-gcc12
     runs-on: ubuntu-latest
     container: ghcr.io/r-hub/containers/${{ matrix.image }}:latest
     steps:
@@ -320,6 +320,7 @@ jobs:
                 qpdf \
                 texinfo \
                 texinfo-tex \
+                texlive-collection-fontsrecommended \
                 texlive-latex \
                 tidy
           fi

--- a/.github/workflows/r_package.yml
+++ b/.github/workflows/r_package.yml
@@ -274,13 +274,12 @@ jobs:
         #   * CRAN "additional checks": https://cran.r-project.org/web/checks/check_issue_kinds.html
         #   * images: https://r-hub.github.io/containers/containers.html
         image:
+          # unless a specific reason to do otherwise, just test
+          # oldest and newest supported versions of compilers here
           - clang16
-          - clang17
-          - clang18
-          - clang19
-          - clang20
+          - clang22
           - gcc14
-          - intel
+          - gcc16
           - rchk
     runs-on: ubuntu-latest
     container: ghcr.io/r-hub/containers/${{ matrix.image }}:latest

--- a/.github/workflows/r_package.yml
+++ b/.github/workflows/r_package.yml
@@ -303,6 +303,7 @@ jobs:
             apt-get install --no-install-recommends -y \
                 cmake \
                 devscripts \
+                libcurl4-openssl-dev \
                 libuv1-dev \
                 texinfo \
                 texlive-latex-extra \
@@ -316,6 +317,7 @@ jobs:
             yum install -y \
                 cmake \
                 devscripts \
+                libcurl-devel \
                 libuv-devel \
                 qpdf \
                 texinfo \
@@ -358,7 +360,11 @@ jobs:
           #   - 'curl' is not a dependency of {lightgbm}, but it's needed for 'R CMD check' URL checks
           #   - 'testthat' is not needed by 'rchk', so avoid installing it until here
           #
-          Rscript -e "install.packages(c('curl', 'testthat'), repos = 'https://cran.rstudio.com', Ncpus = parallel::detectCores())"
+          Rscript -e "
+            install.packages(c('curl', 'testthat'),
+            repos = 'https://cran.rstudio.com',
+            Ncpus = parallel::detectCores())
+          "
 
           if [[ "${{ matrix.image }}" =~ "clang" ]]; then
             # allowing the following NOTEs (produced by default in the clang images):

--- a/.github/workflows/r_package.yml
+++ b/.github/workflows/r_package.yml
@@ -302,6 +302,7 @@ jobs:
             apt-get update
             apt-get install --no-install-recommends -y \
                 cmake \
+                curl \
                 devscripts \
                 libuv1-dev \
                 texinfo \
@@ -315,12 +316,16 @@ jobs:
             yum update -y
             yum install -y \
                 cmake \
+                curl \
                 devscripts \
                 libuv-devel \
                 qpdf \
                 texinfo \
                 texinfo-tex \
+                texlive-amsmath \
+                texlive-amsfonts \
                 texlive-collection-fontsrecommended \
+                texlive-collection-latexrecommended \
                 texlive-latex \
                 tidy
           fi

--- a/.github/workflows/r_package.yml
+++ b/.github/workflows/r_package.yml
@@ -302,7 +302,6 @@ jobs:
             apt-get update
             apt-get install --no-install-recommends -y \
                 cmake \
-                curl \
                 devscripts \
                 libuv1-dev \
                 texinfo \
@@ -316,7 +315,6 @@ jobs:
             yum update -y
             yum install -y \
                 cmake \
-                curl \
                 devscripts \
                 libuv-devel \
                 qpdf \
@@ -355,8 +353,12 @@ jobs:
             fi
           fi
 
-          # 'testthat' is not needed by 'rchk', so avoid installing it until here
-          Rscript -e "install.packages('testthat', repos = 'https://cran.rstudio.com', Ncpus = parallel::detectCores())"
+          # why these packages?
+          #
+          #   - 'curl' is not a dependency of {lightgbm}, but it's needed for 'R CMD check' URL checks
+          #   - 'testthat' is not needed by 'rchk', so avoid installing it until here
+          #
+          Rscript -e "install.packages(c('curl', 'testthat'), repos = 'https://cran.rstudio.com', Ncpus = parallel::detectCores())"
 
           if [[ "${{ matrix.image }}" =~ "clang" ]]; then
             # allowing the following NOTEs (produced by default in the clang images):

